### DISCUSSION TITLE
Failed Tests in outputs/ and meshgenerators/ and some warnings during compilation

### DISCUSSION THREAD

**Discussion by smpeyres** [idaholab/moose#25488] [Q&A Getting Started]
Dear MOOSE Team, I updated my moose repository but when I re-compiled I got a set of warnings, which all seemed to revolve around my use of the M1 Mac. I usually ignore these warnings if all of the tests pass, but this time a handful within outputs/ and meshgenerators/ do fail, all with the error of EXODIFF. The full set of warnings and the details of the EXODIFF errors are too big to copy and pasted, so I attached a couple of .txt files with the entirety of the compilation and test results.

Any advice?

Thanks!
[compile_warnings.txt](https://github.com/idaholab/moose/files/12643489/compile_warnings.txt)
[test_results.txt](https://github.com/idaholab/moose/files/12643490/test_results.txt)

**Comment by GiudGiud** [comment]
The warnings at link time are normal. It's some issue with compilation visibility setttings that get reported by newer compilers. 
We all get them. We haven't gotten to fixing that, there's always something more urgent to tackle.

The failures are all tied to Nemesis output
```
meshgenerators/output_intermediate_mesh.intermediate_mesh_output/nemesis [90m.......[0m [33m[min_cpus=2][0m [33mFAILED (EXODIFF)[0m
outputs/variables.nemesis_hide [90m.................................................[0m [33m[min_cpus=2][0m [33mFAILED (EXODIFF)[0m
meshgenerators/distributed_rectilinear/dmg_displaced_mesh.pbc/adaptivity_nemesis [33m[min_cpus=4][0m [33mFAILED (EXODIFF)[0m
outputs/nemesis.nemesis_elemental_replicated [90m...................................[0m [33m[min_cpus=4][0m [33mFAILED (EXODIFF)[0m
outputs/nemesis.nemesis_scalar_replicated [90m......................................[0m [33m[min_cpus=4][0m [33mFAILED (EXODIFF)[0m
```

we have JUST done a libmesh submodule update that changes the outcome of these tests. If you update MOOSE without updating libmesh you ll get those failures. If you update libmesh they will go away

Guillaume

**Reply by smpeyres** [reply]
I ran 'mamba update --all' but the same tests still fail. Do I need to run something like 'git submodule update --init libmesh' in order to update libmesh? Update: I tried this as well, to no avail.

**Reply by GiudGiud** [reply]
did you use conda/mamba to install moose?
If so then you should not use the libmesh submodule

btw the mamba update may be failing these days due to dependency resolution issues. See this post:
https://github.com/idaholab/moose/discussions/25358

**Reply by smpeyres** [reply]
Yes I did use mamba to install moose (following the basic Linus/MacOS MOOSE Installation guide). 

Do you suggest I perform 'mamba install moose-dev=2023.08.29' for this?

**Reply by GiudGiud** [reply]
let's do this yes

**Reply by smpeyres** [reply]
I made a fresh clone of MOOSE without the libmesh update and tried the old version of moose-dev, but the same five tests still fail.

Any suggestions?

**Reply by GiudGiud** [reply]
new moose + old libmesh -> expect failure
old moose (pre libmesh update) + new libmesh -> expect failure

These failures are understood and are only related to Nemesis output. Unless you are working with that, I would just move on and expect it to work again as expected the next time you update moose and libmesh

**Reply by smpeyres** [reply]
I am not using Nemesis, so I'll move on for now... thanks!
